.github/workflows/CI-Operator.yml

```diff
@@ -0,0 +1,149 @@
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ env.TESTING_FRAMEWORK_REPO }}
+          path: testing-framework
+          ref: ${{ needs.create-test-ref.outputs.testRef }}
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: '~1.17.7'
+
+      - name: Create test batch key values
+        id: set-batches
+        run: |
+          cd testing-framework/tools/batchTestGenerator
+          go build
+          ./batchTestGenerator github --testCaseFilePath=./testcases.json --maxBatch=${{ env.NUM_BATCHES }} \
+            --eksarm64amp=${{ env.EKS_ARM_64_AMP_ENDPOINT }} \
+            --eksarm64region=${{ env.EKS_ARM_64_REGION }} \
+            --eksarm64cluster=${{ env.EKS_ARM_64_CLUSTER_NAME }} \
+            --include=EKS_ADOT_OPERATOR
+      - name: List testing suites
+        run: |
+          echo ${{ steps.set-batches.outputs.batch-keys }}
+          echo ${{ steps.set-batches.outputs.batch-values }}
+
+  run-batch-job:
+    runs-on: ubuntu-latest
+    needs: [get-testing-suites, create-test-ref]
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.get-testing-suites.outputs.test-case-batch-key) }}
+
+    steps:
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+
+      - name: Set up terraform
+        uses: hashicorp/setup-terraform@v2
+
+      - name: Configure AWS Credentials
+        if: steps.e2etest-eks.outputs.cache-hit != 'true'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.INTEG_TEST_AWS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.INTEG_TEST_AWS_KEY_SECRET }}
+          aws-region: us-west-2
+      
+      - name: Checkout testing framework
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ env.TESTING_FRAMEWORK_REPO }}
+          path: testing-framework
+          ref: ${{ needs.create-test-ref.outputs.testRef }}
+
+      - name: create test-case-batch file
+        run: |
+          jsonStr='${{ needs.get-testing-suites.outputs.test-case-batch-value }}'
+          jsonStr="$(jq -r '.${{ matrix.BatchKey }} | join("\n")' <<< "${jsonStr}")"
+          echo "$jsonStr" >> testing-framework/terraform/test-case-batch
+          cat testing-framework/terraform/test-case-batch
+
+      - name: Get TTL_DATE for cache
+        id: date
+        run: echo "::set-output name=ttldate::$(date -u -d "+7 days" +%s)"
+
+      # version used is the same as determined during 'e2etest-preparation' to ensure same 'adot-collector-integration-test' image is used
+      - name: Versioning for testing
+        id: versioning
+        run: |
+          version="$(cat VERSION)-$(git rev-parse --short HEAD)"
+          echo $version > $PACKAGING_ROOT/VERSION
+          cat $PACKAGING_ROOT/VERSION
+          echo "::set-output name=version::$version"
```

Comment:
This creates a direct dependency on the Collector CI being ran so that this image is created. Is this a coupling we want to accept? 

Also, don't we want to guarantee that we are testing upstream collector version against operator version. This versioning strategy does not guarantee that.

Reply:
Wouldn't we want to test against the downstream collector version that is being released, i.e. this image that's being created by Collector CI? Are we releasing the Operator on a different cadence than the Collector?

Reply:
I believe this could be okay, as the CI for releases would be done on a release branch and I do not think collector version drift would be an issue there. I do wonder if there are edge cases we haven't quite considered.